pandas/core/internals/blocks.py

```diff
@@ -1771,6 +1772,19 @@ def _slice(self, slicer):
 
         return self.values[slicer]
 
+    def _try_cast_result(self, result, dtype=None):
+        """
+        if we have an operation that operates on for example floats
+        we want to try to cast back to our EA here if possible
+        """
+        try:
+            result = self._holder._from_sequence(
+                np.asarray(result).ravel(), dtype=dtype)
```

Comment:
Why is the `asarray` and ravel needed? `_from_sequence` should take any sequence, so the trip through NumPy shouldn't be needed. Can `result` be 2-D here? I can't imagine us wanting to ravel a 2-D array, since the length will change.

Reply:
I added some comments (short answer is this *could* be a 2-D numpy array or a EA which is already 1-D)